models/dc_generator.cpp

```diff
@@ -111,12 +165,30 @@ nest::dc_generator::calibrate()
 void
 nest::dc_generator::update( Time const& origin, const long from, const long to )
 {
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+
   long start = origin.get_steps();
 
   CurrentEvent ce;
   ce.set_current( P_.amp_ );
 
   for ( long offs = from; offs < to; ++offs )
+  {
+    B_.logger_.record_data( origin.get_steps() + offs );
```

Comment:
Was wondering if the same should also be true for 'sinusodial_poisson_generator.cpp', line 268:
https://github.com/nest/nest-simulator/blob/master/models/sinusoidal_poisson_generator.cpp#L268